editor/sidebar/featured-image/index.js

```diff
@@ -52,19 +53,18 @@ class FeaturedImage extends Component {
 			return;
 		}
 		this.setState( { loading: true } );
-		const mediaIdToLoad = this.props.featuredImageId;
-		this.fetchMediaRequest = new wp.api.models.Media( { id: mediaIdToLoad } ).fetch()
+		if ( this.fetchMediaRequest ) {
+			this.fetchMediaRequest.abort();
+		}
+		this.fetchMediaRequest = new wp.api.models.Media( { id: this.props.featuredImageId } ).fetch()
 			.done( ( media ) => {
-				if ( this.props.featuredImageId !== mediaIdToLoad ) {
-					return;
-				}
 				this.setState( {
 					loading: false,
 					media,
 				} );
 			} )
-			.fail( () => {
-				if ( this.props.featuredImageId !== mediaIdToLoad ) {
+			.fail( ( error ) => {
+				if ( error && error.statusText === 'abort' ) {
```

Comment:
Looking to the docs, the first argument is not the error, but the jqXHR instance:

http://api.jquery.com/jQuery.ajax/#jqXHR

Is it ever falsey?